File: apps/api/src/application/orders/get-order.ts
import { NotFoundError } from '@/domain/shared/domain-error'

import type { OrderRepository } from '@/application/ports/order-repository'
import type { Order } from '@/domain/order/order'

export class GetOrder {
  constructor(private readonly orders: OrderRepository) {}

  async execute(id: string): Promise<Order> {
    const order = await this.orders.findById(id)
    if (!order) {
      throw new NotFoundError('order', id)
    }
    return order
  }
}

File: apps/api/src/ecommerce_api/application/orders/get_order.py
from ecommerce_api.application.ports.order_repository import OrderRepository
from ecommerce_api.domain.errors import NotFoundError
from ecommerce_api.domain.order import Order


class GetOrder:
    def __init__(self, orders: OrderRepository) -> None:
        self._orders = orders

    async def execute(self, id: str) -> Order:
        order = await self._orders.find_by_id(id)
        if order is None:
            raise NotFoundError('order', id)
        return order
